app-web/__tests__/pages/Index.test.js

```diff
@@ -1,116 +1,142 @@
-    resourcesByType: groupBy(SIPHON_NODES, 'resource.type'),
-    searchResultsExist: false,
-    location,
-    ...actions,
-  };
+  test('it matches snapshot, when there is a search and no results an alert box shows up', () => {
+    queryString.parse.mockReturnValue({});
+    const { container, rerender, queryByTestId } = render(
+      <ThemeProvider theme={theme}>
+        <Index {...props} />
+      </ThemeProvider>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+    let Alert;
+    Alert = queryByTestId(TEST_IDS.alert);
+    // initially there are valid results and so there should no be no alert
+    expect(Alert).not.toBeInTheDocument();
+    // mock that query string actually finds search values in the url
+    queryString.parse.mockReturnValue({ q: 'foo' });
+    // rerender stubbing no results for resources
+    useSearch.mockReturnValue([]);
+    rerender(
+      <ThemeProvider theme={theme}>
+        <Index {...props} />
+      </ThemeProvider>,
+    );
+    expect(useSearch).toHaveBeenCalled();
+    Alert = queryByTestId(TEST_IDS.alert);
 
-  test('it matches snapshot', () => {
-    const wrapper = shallow(<Index {...props} />);
-    expect(wrapper).toMatchSnapshot();
+    expect(Alert).toBeInTheDocument();
   });
 
-  describe('Lifecycle methods', () => {
-    test('calls load resources', () => {
-      shallow(<Index {...props} />);
-      expect(actions.loadResources).toHaveBeenCalled();
-    });
+  test('when there is an empty search the alert box does not show and all results show instead', () => {
+    queryString.parse.mockReturnValue({});
+    const { container, rerender, queryByTestId, getByTestId } = render(
+      <ThemeProvider theme={theme}>
+        <Index {...props} />
+      </ThemeProvider>,
+    );
+    expect(container.firstChild).toMatchSnapshot();
+    let Alert;
+    Alert = queryByTestId(TEST_IDS.alert);
+    // initially there are valid results and so there should no be no alert
+    expect(Alert).not.toBeInTheDocument();
+    // mock that query string actually finds search values in the url
+    queryString.parse.mockReturnValue({ q: '' });
+    // rerender stubbing no results for resources
+    useSearch.mockReturnValue([]);
+    rerender(
+      <ThemeProvider theme={theme}>
+        <Index {...props} />
+      </ThemeProvider>,
+    );
+    expect(useSearch).toHaveBeenCalled();
+    Alert = queryByTestId(TEST_IDS.alert);
+
+    expect(Alert).not.toBeInTheDocument();
 
-    test('when search props updates, it calls search function', async () => {
-      const wrapper = shallow(<Index {...props} />);
-      wrapper.setProps({ location: { ...location, search: '?q=foo' } });
-      await wrapper.instance().componentDidMount();
-      expect(getSearchResults).toHaveBeenCalled();
-      expect(props.setSearchQuery).toHaveBeenCalled();
-      expect(props.setSearchResults).toHaveBeenCalled();
-    });
+    expect(getByTestId(COLLECTION_TEST_IDS.container)).toBeInTheDocument();
```

Comment:
If `getByTestId` throws, reconsider on asserting it.

Reply:
changed to query by test id which does not throw